security/oc_store.c

```diff
@@ -29,16 +29,19 @@
 #include "port/oc_storage.h"
 #include <oc_config.h>
 
-#ifdef OC_APP_DATA_STORAGE_BUFFER
-static uint8_t buf[OC_APP_DATA_BUFFER_SIZE];
-#endif /* OC_APP_DATA_STORAGE_BUFFER */
-
 #ifdef OC_DYNAMIC_ALLOCATION
 #include <stdlib.h>
 #else /* OC_DYNAMIC_ALLOCATION */
 #define OC_APP_DATA_STORAGE_BUFFER
+#ifndef OC_APP_DATA_BUFFER_SIZE
+#define OC_APP_DATA_BUFFER_SIZE OC_MAX_APP_DATA_SIZE
+#endif /* OC_APP_DATA_BUFFER_SIZE */
 #endif /* !OC_DYNAMIC_ALLOCATION */
 
+#ifdef OC_APP_DATA_STORAGE_BUFFER
+static uint8_t buf[OC_APP_DATA_BUFFER_SIZE];
```

Comment:
`buf` seems like a dangerous name for a file-static variable. 

I guess it's a deliberate shadowing with the malloc-ed buf when `OC_APP_DATA_STORAGE_BUFFER`, '-Wshadow' is enabled thus the compiler should catch shadowing problems, so I guess it's alright.

Reply:
added check  `-Wshadow` to build